core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -273,24 +283,35 @@ public boolean testClassLoad(final String className, final String asTypeName)
   @Override
   public List<ActiveCompaction> getActiveCompactions(String tserver)
       throws AccumuloException, AccumuloSecurityException {
-    final var parsedTserver = HostAndPort.fromString(tserver);
-    Client client = null;
-    try {
-      client = getClient(ThriftClientTypes.TABLET_SERVER, parsedTserver, context);
+    final var serverHostAndPort = HostAndPort.fromString(tserver);
 
-      List<ActiveCompaction> as = new ArrayList<>();
-      for (var tac : client.getActiveCompactions(TraceUtil.traceInfo(), context.rpcCreds())) {
-        as.add(new ActiveCompactionImpl(context, tac, parsedTserver, CompactionHost.Type.TSERVER));
+    final List<ActiveCompaction> as = new ArrayList<>();
+    try {
+      if (getTabletServers().contains(tserver)) {
```

Comment:
For code like the following with a large number of compactors and a large number of tservers the call `getTabletServers().contains(tserver)` could get really expensive because the getTabletSevers call returns a List.  If it returned a set it would be slightly better, but still expensive to recreate the set per compactor.

```java
AccumuloClient client = ...;

for(var compactor : client.instanceOperations().getCompactors()) {
    for(var compaction : client.instanceOperations().getActiveCompactions(compactor){
      System.out.println(compaction);
   }
}
```

Reply:
Not sure if it possible because it depends on the ZK schema.  Wondering if we could just see if the specific tserver exist in zoocache.

Reply:
Found some code that I could reuse. I added it to ZookeeperLockChecker and wired it up to ClientContext. Please take a look at TabletLocator changes as this would re-use the existing ZookeeperLockChecker instance instead of creating a new one each time.